library/bigip_software_facts.py

```diff
@@ -0,0 +1,306 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {
+    'status': ['preview'],
+    'supported_by': 'community',
+    'metadata_version': '1.0'
+}
+
+DOCUMENTATION = '''
+---
+module: bigip_software_facts
+short_description: Collect software facts from BIG-IP devices.
+description:
+  - Collect information about installed volumes, existing ISOs for images and hotfixes on the BIG-IP device.
+version_added: "2.5"
+options:
+  include:
+    description:
+      - Type of information to collect.
+    required: False
+    default: ['all']
+    choices:
+      - all
+      - image
+      - hotfix
+      - volume
+  filter:
+    description:
+      - Filter responses based on the attribute and value provided. Valid filters are required to be in 
+      C(key:value) format, with keys being one of the following: name, build, version, status, active.
+notes:
+   - Requires the f5-sdk Python package on the host. This is as easy as
+     pip install f5-sdk
+extends_documentation_fragment: f5
+requirements:
+  - f5-sdk
+author:
+  - Wojciech Wypior (@wojtek0806)
+'''
+
+EXAMPLES = '''
+- name: Gather image facts filter on version
+  bigip_software_facts:
+      server: "lb.mydomain.com"
+      user: "admin"
+      password: "secret"
+      include: "image"
+      filter: "version:12.1.1"
+  delegate_to: localhost
+'''
+
+RETURN = '''
+images:
+    description:
+        List of base image ISOs that are present on the unit.
+    returned: changed
+    type: list of dict
+    sample:
+        images:
+            - build: "0.0.184"
+              fileSize: "1997 MB",
+              lastModified: "Sun Oct  2 20:50:04 2016",
+              name: "BIGIP-12.1.1.0.0.184.iso",
+              product: "BIG-IP",
+              version: "12.1.1"
+hotfixes:
+    description:
+        List of hotfix ISOs that are present on the unit.
+    returned: changed
+    type: list of dict
+    sample:
+        hotfixes:
+            - build: "2.0.204"
+              fileSize: "1997 MB",
+              lastModified: "Sun Oct  2 20:50:04 2016",
+              name: "12.1.1-hf2.iso",
+              product: "BIG-IP",
+              version: "12.1.1"   
+volumes:     
+    description:
+        List the volumes present on device.
+    returned: changed
+    type: list of dict
+    sample:
+       volumes:
+            - basebuild: "0.0.184",
+              build: "0.0.184",
+              name: "HD1.2",
+              product: "BIG-IP",
+              status: "complete",
+              version: "12.1.1"              
+'''
+
+
+from ansible.module_utils.f5_utils import (
+    AnsibleF5Client,
+    AnsibleF5Parameters,
+    HAS_F5SDK,
+    F5ModuleError,
+)
+
+
+class Parameters(AnsibleF5Parameters):
+    returnables = ['name', 'build', 'version',
```

Comment:
Please start this list on a newline. And end the list so that it is aligned with the variable name.
ex.
```
var1 = [
    'foo'
]
```